### DISCUSSION TITLE
What "standard" aspect ratios are supported for birdseye in 0.13?

### DISCUSSION THREAD

**Discussion by natevw** [blakeblackshear/frigate#8580] [Ask A Question] [answered]
I had my birdseye set up with:

```
birdseye:
  mode: continuous
  width: 1920
  height: 1200
```

under 0.12.x but upon trying out the 0.13 beta 5 release I started getting a crazy glitchy rolling feed there, basically like the FAQ item <https://docs.frigate.video/troubleshooting/faqs#my-mjpeg-stream-or-snapshots-look-green-and-crazy> but for the Birdseye rather than a particular camera ffmpeg configuration like that discusses.

I did find back  <https://github.com/blakeblackshear/frigate/discussions/8366#discussioncomment-7429179> which says:

> your birdseye resolution is not using a standard aspect ratio

and later downthread:

> But will need to force resolutions to fit standard aspect ratios

**Which** aspects ratio(s) are considered "standard"? I gather I should be able to still bump up the size/resolution but I just need to make sure that the values are some particular ratio… 4:3? 16:9? (after many evenings of tinkering and troubleshooting I'm getting kind of burned out on waiting for Frigate to reboot so it'd be nice to just know rather than trying to guess…)

**Comment by NickM-27** [comment, marked as answer]
This will be added to the docs but basically the answer is the vertical and horizontal versions of 16x9 4:3 2:1 32:9

**Reply by natevw** [reply]
Ok, thanks. For now I just doubled each width+height value shown as the default in the config docs. (Is the layout algorithm still being worked on? Hate to complain too much, but it just doesn't seem to use the available space at all optimally, especially with `continuous` it basically crams everything into the top left corner — I did recently add a couple cameras so maybe that throws it off, but overall it seems a lot of downsides and limitations compared to the 0.12 version.)

**Comment by riffy13** [comment]
This seems to have gone live. So, what about 1:1 cameras like fisheye? Frigate is not liking my reolink fisheyes that are 2560x2560. Flashing green if I don't use one of those ratios. When I do... my five cameras are just lined up along the top, not using the space well. On the plus side, they stay in order now. :-)

**Reply by faqall** [reply]
Although my cameras are not fisheye, I am experiencing the same issue. It is less than ideal to have them all lined up in the top. If I don't use a standard ratio birds eye goes haywire! 

Note: My camera streams for birds eye are each 640x720. 

https://github.com/blakeblackshear/frigate/issues/9655

![image](https://github.com/blakeblackshear/frigate/assets/74998598/55e74701-2461-4df5-aa89-bfce6a9b87ae)

**Reply by NickM-27** [reply]
There's no way to fit all those cameras in 16:9 while they are all the same size. What you are showing is the optimal layout given the parameters. This is also how they would be lined up in the grid in 0.12

You could try using a 4:3 aspect ratio instead and see how that lines up

**Reply by faqall** [reply]
I guess what I am saying is in .12 I could set birds eye view to whatever resolution I wanted to without an issue. I just pinned my container to .12 so I could show you how I have my 1340x1440 height. 

![image](https://github.com/blakeblackshear/frigate/assets/74998598/403895cf-7e85-4b49-bd7f-d72831de6dbd)

This layout is far more optimal for my config however I can no longer achieve this in .13. 
![image](https://github.com/blakeblackshear/frigate/assets/74998598/3571216f-d9bd-4db3-b891-10bdce404244)

**Reply by faqall** [reply]
And if I set it to 1280x720 in .12 it would at least put them side by side instead of all on one row like .13.

![image](https://github.com/blakeblackshear/frigate/assets/74998598/4fa5a76f-b659-44d6-9e94-088b9ba0677b)

And finally if I set it to 1280x1440 which is perfect for my config. It works great in .12 but in .13 this isnt possible without the glitching green problem. 

![image](https://github.com/blakeblackshear/frigate/assets/74998598/a71ff54a-9666-4d53-8a0d-a7d75751c389)

**Reply by NickM-27** [reply]
Right, it doesn't use a grid anymore. In 0.12 version it uses the same space for each camera as the current implementation just laid out differently. 

You could try making the birdseye layout 3x4 or 9x16 and it might work better

**Reply by faqall** [reply]
I appreciate your responses and guidance! When you say 3x4 or 9x16 are you saying in aspect ratio? Can I get and example in pixels so I can ensure I am following you properly!? Happy to give it a go! Thanks again!

**Reply by NickM-27** [reply]
960x1280 or 720x1280

**Reply by faqall** [reply]
960x1280 cant seem to properly fit all 4
![image](https://github.com/blakeblackshear/frigate/assets/74998598/02b75ac9-eba8-4461-a394-0589d44fdc83)

720x1280 just squeezes them all in a row with a ton of blank space below
![image](https://github.com/blakeblackshear/frigate/assets/74998598/67ae584c-1c62-4fae-be5a-c432276a4b41)

thanks again!

**Comment by riffy13** [comment]
Yeah, my 1:1 filled the space when 4 of 5 where active but now they just stay in a line on top that's to small to use. This is what I get, just a line on top:
<a href="https://imgur.com/GGAxvSj"><img src="https://i.imgur.com/GGAxvSj.png" title="source: imgur.com" /></a>

**Comment by riffy13** [comment]
Still messing with this... What's the new 
"birdseye: layout:  scaling_factor: 2.0" 
setting on the docs? Tried it but get error 
"1 validation error for FrigateConfig birdseye -> layout extra fields not permitted (type=value_error.extra)"

edit: also can't get the listed 2:1 ratio to work

**Reply by NickM-27** [reply]
You are probably not running the correct version, that feature will come as part of 0.13.2

**Reply by riffy13** [reply]
This seems to have gone live now, but no mater what scale I use, I have all cameras lined up on top.
